src/prefect/task_worker.py

```diff
@@ -35,21 +36,28 @@
 from prefect.utilities.annotations import NotSet
 from prefect.utilities.asyncutils import asyncnullcontext, sync_compatible
 from prefect.utilities.engine import emit_task_run_state_change_event
-from prefect.utilities.processutils import _register_signal
+from prefect.utilities.processutils import (
+    _register_signal,  # pyright: ignore[reportPrivateUsage]
+)
 from prefect.utilities.services import start_client_metrics_server
 from prefect.utilities.urls import url_for
 
 logger = get_logger("task_worker")
 
+P = ParamSpec("P")
+R = TypeVar("R")
```

Comment:
Careful, returned types generally have different variance rules from input types.

It is easier to leave figuring out the rules to the type checker by letting it infer them. Import the TypeVar from typing_extensions and then use:

```suggestion
R = TypeVar("R", infer_variance=True)
```

See https://typing.readthedocs.io/en/latest/spec/generics.html#variance-inference